atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+        if not system_info:
+            self.logger.error('No particle information found in H5MD file.')
+
+        n_frames = len(system_info.get('times', []))
+        self._system_time_map = {}
+        self._system_step_map = {}
+        for frame in range(n_frames):
+            # if (n % self.frame_rate) > 0:
+            #     continue
+
+            atoms_dict = {}
+            for key in system_info.keys():
+                if key == 'times':
+                    time = system_info.get('times', [None] * (frame + 1))[frame]
+                    if time is not None:
+                        self._system_time_map[round(ureg.convert(
+                            time.magnitude, time.units, ureg.picosecond), 5)] = len(self._system_time_map)
+                elif key == 'steps':
+                    step = system_info.get('steps', [None] * (frame + 1))[frame]
+                    if step is not None:
+                        self._system_step_map[round(step)] = len(self._system_step_map)
+                else:
+                    atoms_dict[key] = system_info.get(key, [None] * (frame + 1))[frame]
+
+            if frame == 0:  # TODO extend to time-dependent bond lists
+                connectivity = self.h5md_groups['connectivity']
+                atoms_dict['bond_list'] = self._data_parser.hdf5_getter(connectivity, 'bonds')
+
+            self.parse_trajectory_step({
+                'atoms': atoms_dict
+            })
+
+            if frame == 0:  # TODO extend to time-dependent topologies
+                topology = self._data_parser.hdf5_getter(connectivity, 'particles_group', None)
+                if topology:
+                    self.get_atomsgroup_fromh5md(sec_run.system[frame], topology)
+
+    def parse_method(self):
+
+        sec_method = self.archive.run[-1].m_create(Method)
+        sec_force_field = sec_method.m_create(ForceField)
+        sec_model = sec_force_field.m_create(Model)
+
+        # get the atom parameters
+        n_atoms = self.n_atoms[0]  # TODO Extend to non-static n_atoms
+        for n in range(n_atoms):
+            sec_atom = sec_method.m_create(AtomParameters)
+            for key in self.atom_parameters.keys():
+                setattr(sec_atom, key, self.atom_parameters[key][n])
+
+        # Get the interactions
+        connectivity = self.h5md_groups['connectivity']
+        if not connectivity:
+            return
+
+        atom_labels = self.atom_parameters.get('label')
+        interaction_keys = ['bonds', 'angles', 'dihedrals', 'impropers']
+        interactions_by_type = []
+        for interaction_key in interaction_keys:
+            interaction_list = self._data_parser.hdf5_getter(connectivity, interaction_key)
+            if interaction_list is None:
+                continue
+            elif type(interaction_list) == h5py.Group:
+                self.logger.warning('Time-dependent ' + key + ' currently not supported. ' + key + ' list will not be stored')
```

Comment:
remove variables in warning simply: 'Time-dependent key not currently supported, will not be stored.'

Reply:
yes, sorry, I did this prior to our previous discussion about variables within warnings, and did not see this. removed.